kyaml/kio/filters/merge3.go

```diff
@@ -109,13 +110,21 @@ func (m Merge3) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
 type tuples struct {
 	list []*tuple
 
-	// mergeOnPath if set to true will use the resource filepath
-	// as part of the merge key
-	mergeOnPath bool
+	// matcher matches the resources for merge
+	matcher ResourceMatcher
+}
+
+// DefaultMatcher holds the default matching of resources implementation
+type DefaultMatcher struct {
```

Comment:
Can we find a name here that is more descriptive for how resources are matched with this implementation (without it being too long)?

Reply:
Changed to DefaultGKNNMatcher